internal/processor/parser/app.go

```diff
@@ -2,37 +2,70 @@ package parser
 
 import (
 	"fmt"
+	"net/url"
 	"sync"
 	"time"
 
-	"encoding/json"
 	"github.com/DataDog/datadog-firehose-nozzle/internal/metric"
 	"github.com/DataDog/datadog-firehose-nozzle/internal/util"
 	"github.com/cloudfoundry-community/go-cfclient"
 	"github.com/cloudfoundry/gosteno"
 	"github.com/cloudfoundry/sonde-go/events"
-	bolt "github.com/coreos/bbolt"
 )
 
-var clearCacheDuration = 60
+type appCache struct {
+	apps map[string]*App
+	lock sync.RWMutex
+}
+
+func newAppCache() appCache {
+	return appCache{
+		apps: make(map[string]*App),
+	}
+}
+
+func (c *appCache) CreateOrUpdate(resolvedApp cfclient.App) *App {
```

Comment:
why `resolvedApp` and not just `app`?

Reply:
Let's change this to `Add` instead of `CreateOrUpdate`.

Reply:
I used resolvedApp everywhere it's the type `cfclient.App` to differentiate with our own `App` type. I can definitely change the name though, but I want to avoid using just `app`.